crates/bevy_pbr/src/render/fog.wgsl

```diff
@@ -0,0 +1,29 @@
+#define_import_path bevy_pbr::fog
+
+// Fog formulas adapted from:
+// https://learn.microsoft.com/en-us/windows/win32/direct3d9/fog-formulas
+// https://catlikecoding.com/unity/tutorials/rendering/part-14/
+
+fn linear_fog(
+    distance: f32,
+) -> vec4<f32> {
+    var result = fog.color;
+    result.a *= 1.0 - clamp((fog.end - distance) / (fog.end - fog.density_or_start), 0.0, 1.0);
+    return result;
+}
+
+fn exponential_fog(
+    distance: f32,
+) -> vec4<f32> {
+    var result = fog.color;
+    result.a *= 1.0 - 1.0 / exp(distance * fog.density_or_start);
+    return result;
+}
+
+fn exponential_squared_fog(
+    distance: f32,
+) -> vec4<f32> {
+    var result = fog.color;
+    result.a *= 1.0 - 1.0 / exp(pow(distance * fog.density_or_start, 2.0));
```

Comment:
Is using pow 2.0 here really going to be competitive with another two multiplications? Do wgsl compilers optimize this for you?

Reply:
No idea... If the WGSL doesn't do it, maybe the underlying GLSL/SPIRV/MSL compiler does?

Is there a good way to benchmark this? It probably adds up but hard to isolate in the middle of all of the other stuff the shader is doing

Reply:
Hmm, unsure 🤔 Maybe try stress testing it on a very busy scene like animated foxes and then check the traces? Either way, this was mostly curiosity.